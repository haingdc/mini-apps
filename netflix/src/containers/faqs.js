import React from 'react';
import { Accordion, OptForm } from '../components';
import FAQs from '../fixtures/faqs.json';

export function FaqsContainer () {
  var [openId, setOpenId] = React.useState(-1);
  function handleToggle(id) {
    if (id == openId) {
      setOpenId(-1);
    } else {
      setOpenId(id);
    }
  }
  return (
    <Accordion>
      <Accordion.Title>Frequently Asked Questions</Accordion.Title>
      {FAQs.map(item =>
        <Accordion.Item key={item.id}>
          <Accordion.Header toggleShow={openId == item.id} id={item.id} onToggle={handleToggle}>
            {item.header}
          </Accordion.Header>
          <Accordion.Body toggleShow={openId == item.id}>
            <div>
              {item.body}
            </div>
          </Accordion.Body>
        </Accordion.Item>
      )}
      <OptForm>
        <OptForm.Input placeholder="Email Address" />
        <OptForm.Button>Try it now</OptForm.Button>
        <OptForm.Break />
        <OptForm.Text>Ready to watch? Enter your email to create or restart your membership</OptForm.Text>
      </OptForm>
    </Accordion>
  );
}