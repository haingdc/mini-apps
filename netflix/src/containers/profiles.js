import React from 'react';
import * as ROUTES from '../constants/routes';
import logo from '../logo.svg';
import { Frog, Header, Profiles } from '../components';

export function SelectProfileContainer(props) {
  var { user, setProfile } = props;
  return (
    <>
      <Header bg={false}>
        <Header.Frame>
          <Header.Group>
            <Header.Logo to={ROUTES.HOME} src={logo} alt="Netflix" />
            <Header.TextLink> <Frog.StartButton /> </Header.TextLink>
            <Header.TextLink> <Frog.PauseButton /> </Header.TextLink>
          </Header.Group>
        </Header.Frame>
      </Header>
      <Profiles>
        <Profiles.Title>Who's watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.User
            onClick={() => setProfile({ displayName: user.displayName, photoURL: user.photoURL })}
            data-testid="user-profile"
          >
            <Profiles.Picture src={user.photoURL} />
            <Profiles.Name>{user.displayName}</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    </>
  );
}