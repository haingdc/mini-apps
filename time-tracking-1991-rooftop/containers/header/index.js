import Dropdown from '../../components/dropdown/index.js'
import { e } from '../../utils/index.js'

function Header() {
  return (
    e('header', { className: 'app-header' },
      [
        e('img', { key: 'left', src: './assets/images/logo.svg' }),
        e('div', { key: 'right', className: 'right-header' },
          [
            e('img', { key: 'bell', className: 'app__notification', src: './assets/images/notification.svg' }),
            e(Dropdown, { key: 'profile dropdown' },
              [
                e('div', { key: 'profile container', className: 'profile-container' }, [
                  e('img', { key: 'left', src: './assets/images/avatar.png' }),
                  'Adam Fisherman',
                ]),
              ]
            ),
          ]
        ),
      ]
    )
  )
}

export default Header