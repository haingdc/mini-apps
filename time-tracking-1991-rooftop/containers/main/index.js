import { e } from '../../utils/index.js'

function Main() {
  return (
    e('div', undefined,
      [
        e(Section1 , { key: 'left'    }),
        e(Section2 , { key: 'right'   }),
      ]
    )
  )
}

function Section1() {
  return 'Left'
}

var eventsByDay = [
  {
    id: 'Monday',
    events: [
      {
        id: 'ATVPBWQIWR',
        title: 'Tod 1',
        hour: '12:15',
        color: 'blue',
        top: 230,
        height: 50,
      },
      {
        id: 'VAOJCSSKCT',
        title: 'Coding phase',
        hour: '13:45',
        color: 'red',
        top: 330,
        height: 80,
      },
    ],
  },
  {
    id: 'Tuesday',
  },
  {
    id: 'Wednesday',
  },
  {
    id: 'Thursday',
  },
  {
    id: 'Friday',
  },
  {
    id: 'Saturday',
  },
  {
    id: 'Sunday',
  },
]
function Section2() {
  const  days = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']
  const hours = ['1 AM', '2 AM', '3 AM', '4 AM', '5 AM', '6 AM', '7 AM', '8 AM', '9 AM', '10 AM', '11 AM', '12 AM', '13 AM', '14 AM', '15 AM', '16 AM', '17 AM', '18 AM', '19 AM', '20 AM', '21 AM', '22 AM', '23 AM', '24 AM']
  const numbs = [0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48]
  return (
    e('div', { className: 'scheduler' },
      [
        e(
          'div',
          { key: 'hour columns', className: 'column--hour' },
          e(
            'div',
            {},
            [
              e('div', { key: 'gmt', className: 'cell--hour' }, 'GMT+7'),
              hours.map(function renderHour(h) {
                return e(
                  'div',
                  { key: h, className: 'cell--hour cell--14' },
                  e('span', undefined, h)
                )
              })
            ]
          )
        ),
        e('div', { key: 'table', className: 'table' },
          [
            e('div', { key: 'horizontal lines', className: 'horizontal-lines'  },
              numbs.map(function(n, id) {
                return e('div', { key: `horizontal-line-${id}`, className: 'horizontal-line' }, )
              })
            ),
            e('div', { key: 'todo-line', className: 'todo-line' }, ),
            e('div', { key: 'lefty vertical line', className: 'vertical-line vertical-line--lefty' }),
            [
              eventsByDay.map(function renderColumns(day) {
                return e(
                  'div',
                  { key: day.id, className: 'vertical-line' },
                  [
                    e('div', { key: `${day.id}-header`, className: 'cell--header' }, day.id),
                    day?.events?.length
                      ? day.events.map(function renderEvent(event) {
                          return e(
                            Event,
                            {
                              color : event.color,
                              height: event.height,
                              key   : event.id,
                              top   : event.top,
                            },
                            event.title
                          )
                        })
                      : undefined
                  ]
                )
              })
            ]
          ]
        )
      ]
    )
  )
}

export default Main

function Event(props) {
  var { top, height } = props
  const bgrColorLookup = {
    red   : '#FF7976',
    blue  : '#538FFF',
    yellow: '#FEC25A',
  }
  var backgroundColor = bgrColorLookup[props.color] || bgrColorLookup.blue
  return e(
    'div',
    {
      style: {
        zIndex         : 3,
        left           : 0,
        right          : 0,
        top            : top + 'px',
        width          : '80%',
        margin         : '0 auto',
        position       : 'absolute',
        height         : height + 'px',
        color          : '#fff',
        padding        : '7px',
        backgroundColor: backgroundColor,
      }
    },
    props.children
  )
}

function SillyEvent() {
  var content = 'Basgram - code review'
  var hour  = '08:12'
  var code  = 'PR611'
  return e(
    'div',
    {  },
    [
      e('div', { key: 'head', },
        [
          // title
          e('span', { key: 'hour' }, hour),
          // code
          e('span', { key: 'code' }, code),
        ]
      ),
      e('div', { key: 'content' }, content),
    ]
  )
}