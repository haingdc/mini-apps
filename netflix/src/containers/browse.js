import React, { useContext, useEffect, useState } from 'react';
import { SelectProfileContainer } from './profiles';
import { FirebaseContext } from '../context/firebase';
import { Card, Header, Loading } from '../components';
import * as ROUTES from '../constants/routes';
import logo from '../logo.svg';

export function BrowseContainer({ slides }) {
  var [category, setCategory] = useState('series');
  var [searchTerm, setSearchTerm] = useState('');
  var [profile, setProfile] = useState({});
  var [loading, setLoading] = useState(true);
  var [slideRows, setSlideRows] = useState([]);
  var { firebase } = useContext(FirebaseContext);
  var user = firebase.auth().currentUser || {};

  useEffect(() => {
    setTimeout(() => {
      setLoading(false);
    }, 3000);
  }, [profile.displayName]);

  useEffect(() => {
    setSlideRows(slides[category]);
  }, [slides, category]);

  return profile.displayName ? (
    <>
      {loading ? (
        <Loading src={user.photoURL} />
      ) : (
        <Loading.ReleaseBody />
      )}
      <Header src="joker1" dontShowOnSmallViewPort>
        <Header.Frame>
          <Header.Group>
            <Header.Logo to={ROUTES.HOME} alt="Netflix" src={logo} />
            <Header.TextLink
              active={category === 'series' ? 'true' : 'false'}
              onClick={() => setCategory('series')}
            >
              Series
            </Header.TextLink>
            <Header.TextLink
              active={category === 'films' ? 'true' : 'false'}
              onClick={() => setCategory('films')}
            >
              Films
            </Header.TextLink>
          </Header.Group>
          <Header.Group>
            <Header.Search searchTerm={searchTerm} setSearchTerm={setSearchTerm} />
            <Header.Profile>
              <Header.Picture src={user.photoURL} />
              <Header.Dropdown>
                <Header.Group>
                  <Header.Picture src={user.photoURL} />
                  <Header.TextLink>{user.displayName}</Header.TextLink>
                </Header.Group>
                <Header.Group>
                  <Header.TextLink onClick={() => firebase.auth().signOut()}>Sign out</Header.TextLink>
                </Header.Group>
              </Header.Dropdown>
            </Header.Profile>
          </Header.Group>
        </Header.Frame>
        <Header.Feature>
          <Header.FeatureCallOut>Watch Joker Now</Header.FeatureCallOut>
          <Header.Text>Joker is a 2019 American psychological thriller film directed and produced by Todd Phillips, who co-wrote the screenplay with Scott Silver. The film, based on DC Comics characters, stars Joaquin Phoenix as the Joker and provides an alternative origin story for the character. Set in 1981, it follows Arthur Fleck, a failed clown and stand-up comedian whose descent into insanity and nihilism inspires a violent counter-cultural revolution against the wealthy in a decaying Gotham City. Robert De Niro, Zazie Beetz, Frances Conroy, Brett Cullen, Glenn Fleshler, Bill Camp, Shea Whigham, and Marc Maron appear in supporting roles. Joker was produced by Warner Bros. Pictures, DC Films, and Joint Effort, in association with Bron Creative and Village Roadshow Pictures, and distributed by Warner Bros.</Header.Text>
          <Header.PlayButton>Play</Header.PlayButton>
        </Header.Feature>
      </Header>

      <Card.Group>
        {slideRows.map(slideItem => (
          <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
            <Card.Title>{slideItem.title}</Card.Title>
            <Card.Entities>
              {slideItem.data.map(item => (
                <Card.Item key={item.docId} item={item}>
                  <Card.Image src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`} />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              {/* <Player>
                <Player.Button />
                <Player.Video src="/videos/bunny.mp4" />
              </Player> */}
              <p>Hello</p>
            </Card.Feature>
          </Card>
        ))}
      </Card.Group>
    </>
  ) : (
    <SelectProfileContainer user={user} setProfile={setProfile} />
  );
}