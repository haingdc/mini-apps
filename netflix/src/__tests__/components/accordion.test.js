import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import FAQs from  '../../fixtures/faqs.json';
import { FaqsContainer } from '../../containers/faqs';

describe('<Accordion />', () => {
  it('renders the <Accordion /> with populated data', () => {
    var { container, getByText } = render(
      <FaqsContainer />
    );

    expect(getByText('Frequently Asked Questions')).toBeTruthy();
    expect(getByText('What is Netflix?')).toBeTruthy();
    expect(getByText('How much does Netflix cost?')).toBeTruthy();
    expect(getByText('Where can I watch?')).toBeTruthy();
    expect(getByText('How do I cancel?')).toBeTruthy();
    expect(getByText('What can I watch on Netflix?')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('opens and closes the <Accordion /> component', () => {
    var { container, queryByText } = render(
      <FaqsContainer />
    );

    var whatIsNetflixBodyText = "Netflix is a streaming service that offers a wide variety of award-winning TV programmes, films, anime, documentaries and more – on thousands of internet-connected devices.\n\nYou can watch as much as you want, whenever you want, without a single advert – all for one low monthly price. There's always something new to discover, and new TV programmes and films are added every week!";

    console.log(queryByText(whatIsNetflixBodyText))
    expect(queryByText(whatIsNetflixBodyText)).toBeFalsy();
    fireEvent.click(queryByText('What is Netflix?'));
    expect(queryByText(whatIsNetflixBodyText)).toBeTruthy();

    fireEvent.click(queryByText('What is Netflix?'));
    expect(queryByText(whatIsNetflixBodyText)).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});