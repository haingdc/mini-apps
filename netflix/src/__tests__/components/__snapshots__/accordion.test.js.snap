// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> renders the <Accordion /> with populated data 1`] = `
<div
  class="accordion__Container-sbrn4q-0 hXlfWJ"
>
  <div
    class="accordion__Inner-sbrn4q-1 EgRDi"
  >
    <h1
      class="accordion__Title-sbrn4q-6 hLOBnW"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Item-sbrn4q-2 fahRTg"
    >
      <div
        class="accordion__Header-sbrn4q-3 gOnawi"
      >
        What is Netflix?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
      <div
        class="accordion__Body-sbrn4q-4 jGVSmt"
      >
        <div>
          Netflix is a streaming service that offers a wide variety of award-winning TV programmes, films, anime, documentaries and more – on thousands of internet-connected devices.

You can watch as much as you want, whenever you want, without a single advert – all for one low monthly price. There's always something new to discover, and new TV programmes and films are added every week!
        </div>
      </div>
    </div>
    <div
      class="accordion__Item-sbrn4q-2 fahRTg"
    >
      <div
        class="accordion__Header-sbrn4q-3 gOnawi"
      >
        How much does Netflix cost?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
      <div
        class="accordion__Body-sbrn4q-4 jGVSmt"
      >
        <div>
          Watch Netflix on your smartphone, tablet, smart TV, laptop or streaming device, all for one low fixed monthly fee. Plans start from £5.99 a month. No extra costs or contracts.
        </div>
      </div>
    </div>
    <div
      class="accordion__Item-sbrn4q-2 fahRTg"
    >
      <div
        class="accordion__Header-sbrn4q-3 gOnawi"
      >
        Where can I watch?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
      <div
        class="accordion__Body-sbrn4q-4 jGVSmt"
      >
        <div>
          Watch anywhere, anytime, on an unlimited number of devices. Sign in with your Netflix account to watch instantly on the web at netflix.com from your personal computer or on any internet-connected device that offers the Netflix app, including smart TVs, smartphones, tablets, streaming media players and game consoles.

You can also download your favourite programmes with the iOS, Android, or Windows 10 app. Use downloads to watch while you're on the go and without an internet connection. Take Netflix with you anywhere.
        </div>
      </div>
    </div>
    <div
      class="accordion__Item-sbrn4q-2 fahRTg"
    >
      <div
        class="accordion__Header-sbrn4q-3 gOnawi"
      >
        How do I cancel?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
      <div
        class="accordion__Body-sbrn4q-4 jGVSmt"
      >
        <div>
          Netflix is flexible. There are no annoying contracts and no commitments. You can easily cancel your account online in two clicks. There are no cancellation fees – start or stop your account at any time.
        </div>
      </div>
    </div>
    <div
      class="accordion__Item-sbrn4q-2 fahRTg"
    >
      <div
        class="accordion__Header-sbrn4q-3 gOnawi"
      >
        What can I watch on Netflix?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
      <div
        class="accordion__Body-sbrn4q-4 jGVSmt"
      >
        <div>
          Netflix has an extensive library of feature films, documentaries, TV programmes, anime, award-winning Netflix originals, and more. Watch as much as you want, any time you want.
        </div>
      </div>
    </div>
    <div
      class="opt-form__Container-uz9nhu-0 gcnknG"
    >
      <input
        class="opt-form__Input-uz9nhu-1 hfmQUa"
        placeholder="Email Address"
      />
      <button
        class="opt-form__Button-uz9nhu-2 jSmuCg"
      >
        Try it now
        <img
          alt="Try Now"
          src="/images/icons/chevron-right.png"
        />
      </button>
      <div
        class="opt-form__Break-uz9nhu-4 OUzIE"
      />
      <h1
        class="opt-form__Text-uz9nhu-3 cvzEXu"
      >
        Ready to watch? Enter your email to create or restart your membership
      </h1>
    </div>
  </div>
</div>
`;
